src/ResourceManager/Network/Commands.Network/Common/NetworkClient.cs

```
@@ -41,5 +54,177 @@ public NetworkClient(INetworkManagementClient NetworkManagementClient)
+            locationResultsUrl += "?" + string.Join("&", string.Format("api-version={0}", Uri.EscapeDataString(apiVersion)));
+
+            DateTime startTime = DateTime.UtcNow;
+            DateTime giveUpAt = DateTime.UtcNow.AddMinutes(3);
+
+            // Send the Get locationResults request for operaitonId till either we get StatusCode 200 or it time outs (3 minutes in this case)
+            while (true)
+            {
+                HttpRequestMessage newHttpRequest = new HttpRequestMessage();
+                newHttpRequest.Method = new HttpMethod("GET");
+                newHttpRequest.RequestUri = new Uri(locationResultsUrl);
+
+                if (NetworkManagementClient.Credentials != null)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await NetworkManagementClient.Credentials.ProcessHttpRequestAsync(newHttpRequest, cancellationToken).ConfigureAwait(false);
+                }
+
+                HttpResponseMessage newHttpResponse = await httpClient.SendAsync(newHttpRequest, cancellationToken).ConfigureAwait(false);
+
+                if ((int)newHttpResponse.StatusCode != 200)
+                {
+                    if (DateTime.UtcNow > giveUpAt)
+                    {
+                        if (!string.IsNullOrEmpty(newHttpResponse.Content.ReadAsStringAsync().Result))
+                        {
+                            result.Body = newHttpResponse.Content.ReadAsStringAsync().Result;
+                        }
+                        else
+                        {
+                            string newResponseContent = await newHttpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                            throw new Exception(string.Format("Get-AzureRmVpnClientPackage Operation returned an invalid status code '{0}' with Exception:{1} while retrieving " +
```

I think you want to throw an exception whenever you are past the 'giveUpAt' time, so this exception throw belongs outside the else block, right?